src/support/modify.c

```diff
@@ -16,16 +16,15 @@ bool
 __wt_modify_idempotent(const void *modify)
 {
     WT_MODIFY mod;
-    size_t tmp;
-    const size_t *p;
-    int nentries;
+    size_t nentries;
+    const uint8_t *p;
 
     /* Get the number of modify entries. */
     p = modify;
-    memcpy(&tmp, p++, sizeof(size_t));
-    nentries = (int)tmp;
```

Comment:
Do you know why we previously had `nentries` as an int + all the casting? It seems like very strange code but I'm probably missing something. Same question for `__wt_modify_max_memsize`.

Reply:
I presume this was done because of an if check inside `WT_MODIFY_FOREACH_BEGIN`, follow here: 
```
if ((int)__i < (int)napplied)
```
Sometimes `napplied` is 0, and the compiler doesn't like comparing an unsigned variable with 0.

Reply:
That makes sense, thanks.